#!/usr/bin/env python
PACKAGE = "sentry_chassis_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 为8个轮子的PID参数创建配置项（示例：前4个轮子，后4个舵轮同理）
gen.add("Kp_motor", double_t, 0, "PID P gain for motor ", 1.0, 0.0, 20.0)
gen.add("Ki_motor", double_t, 0, "PID I gain for motor ", 0.1, 0.0, 50.0)
gen.add("Kd_motor", double_t, 0, "PID D gain for motor ", 0.0, 0.0, 50.0)
 #I累积的上限，最好等于达到目标速度时的阻力[-1,1]
gen.add("i_clamp_motor", double_t, 0, "PID I clamp for motor", 1.0, 0.0, 5.0)

gen.add("Kp_servo", double_t, 0, "PID P gain for servo ", 5.0, 0.0, 20.0)
gen.add("Ki_servo", double_t, 0, "PID I gain for servo ", 0.1, 0.0, 5.0)
gen.add("Kd_servo", double_t, 0, "PID D gain for servo ", 0.5, 0.0, 10.0)
gen.add("i_clamp_servo", double_t, 0, "PID I clamp forservo", 1.0, 0.0, 5.0)

gen.add("target_motor_all", double_t, 0, "set target v for all motor", 0.0,-50.0, 50.0)
gen.add("target_servo_all", double_t, 0, "set target p for all servo", 0.0, -3.14159,3.14159)#左正右负

exit(gen.generate(PACKAGE, "sentry_chassis_controller", "ChassisPID"))
